src/collective.jl

```diff
@@ -167,6 +167,45 @@ Scatter(sendbuf, T, comm; root::Integer=Cint(0)) =
 Scatter(sendbuf, ::Type{T}, root::Integer, comm::Comm) where {T} =
     Scatter!(sendbuf, Ref{T}(), root, comm)[]
 
+"""
+    scatter(objs::Union{AbstractVector, Nothing}, comm::Comm; root::Integer=0)
+
+Sends the `j`-th element of `objs` in the `root` process to rank `j-1` and returns it. On `root`, `objs` is expected to be a `Comm_size(comm)`-element vector. On the other ranks, it is ignored and can be `nothing`.
+
+This method can handle arbitrary data.
+
+# See also
+
+- [`Scatter!`](@ref)
+"""
+function scatter(objs::Union{AbstractVector, Nothing}, comm::Comm; root::Integer=0)
+    isroot = Comm_rank(comm) == root
+
+    if isroot
+        if length(objs) != Comm_size(comm)
+            throw(ArgumentError("Length of argument objs ($(length(objs))) != number of ranks in comm ($(Comm_size(comm)))."))
+        end
+
+        datasegments = MPI.serialize.(objs)
+        counts = length.(datasegments)
+
+        count = Scatter(counts, Int64, comm; root = root)
+        sendbuf = VBuffer(vcat(datasegments...), counts)
```

Comment:
Splatting vectors is generally a bad idea. 

The optimal approach would be to incrementally serialize to a single buffer, but at the minimum we should call `reduce`:
```suggestion
        sendbuf = VBuffer(reduce(vcat, datasegments), counts)
```